org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/consistencycheck/ConsistencyCheck.java

```diff
@@ -20,4 +36,110 @@ public void checkConsistency(IRepository repository) {
 
 	}
 
+	public List<ConsistencyCheckErrors> checkForConsistency(IRepository repository) {
+		List<ConsistencyCheckErrors> errors = new ArrayList<>();
+
+		errors.addAll(checkConsistencyArtifactTemplate(repository.getAllDefinitionsChildIds(ArtifactTemplateId.class), repository));
+		errors.addAll(checkConsistencyPolicyTemplate(repository.getAllDefinitionsChildIds(PolicyTemplateId.class), repository));
+		errors.addAll(checkConsistencyServiceTemplate(repository.getAllDefinitionsChildIds(ServiceTemplateId.class), repository));
+		
+		errors.addAll(checkConsistencyNodeTypeImplementation(repository.getAllDefinitionsChildIds(NodeTypeImplementationId.class), repository));
+		errors.addAll(checkConsistencyRelationshipTypeImplementation(repository.getAllDefinitionsChildIds(RelationshipTypeImplementationId.class), repository));
+		errors.addAll(checkConsistencyArtifactType(repository.getAllDefinitionsChildIds(ArtifactTypeId.class), repository));
+		
+		errors.addAll(checkConsistencyCapabilityType(repository.getAllDefinitionsChildIds(CapabilityTypeId.class), repository));
+		errors.addAll(checkConsistencyNodeType(repository.getAllDefinitionsChildIds(NodeTypeId.class), repository));
+		errors.addAll(checkConsistencyRequirementType(repository.getAllDefinitionsChildIds(RequirementTypeId.class), repository));
+
+		return errors;
+	}
+
+	private List<ConsistencyCheckErrors> checkConsistencyArtifactTemplate(Set<ArtifactTemplateId> checkForLinkedInRepository, IRepository repository) {
```

Comment:
Variable name `checkForLinkedInRepository` should be named `artifactTemplateIds`. Otherwise, it does not match the method name and the intention of the method.

(Applies for all other methods, too)